R/fit.R

```diff
@@ -937,32 +937,36 @@ CmdStanMCMC <- R6::R6Class(
       private$metadata_ <- data_csv$metadata
 
       if (!is.null(data_csv$post_warmup_draws)) {
+        missing_variables <- !(posterior::variables(data_csv$post_warmup_draws) %in% posterior::variables(private$draws_))
```

Comment:
This fixes an issue that is related to the main issue this PR is fixing. It was a bug all along but only surfaced with this.

```r
library(cmdstanr)
stan_program <- file.path(cmdstan_path(), "examples/bernoulli/bernoulli.stan")
model <- cmdstan_model(stan_program)

stan_data <- list(N = 10, y = c(0,1,0,0,0,0,0,0,0,1))
fit3 = model$sample(data = stan_data, parallel_chains = 4, save_warmup = 1, iter_warmup = 100, iter_sampling = 0, save_latent_dynamics = TRUE)

fit3$draws() #works fine
fit3$draws() #fails

fit3$sampler_diagnostics() #works fine
fit3$sampler_diagnostics() #fails
```